### PR TITLE
Enable customization via Site Editor

### PR DESCRIPTION
**Pull request by victorhmp**
#### What does this PR do? \*

Adds support for customization of certain aspects of a `slider-layout` using our Site Editor.

#### How to test it? \*

The second slider at https://imageslider--storecomponents.myvtex.com/admin/cms/site-editor/about-us can be customized.

#### Related to / Depends on \*

This enables https://github.com/vtex-apps/store-image/pull/3 to be released.

### CODE REVIEW

File: react/components/Slider.tsx [hunk 1]
```diff
@@ -10,22 +10,27 @@ import PaginationDots from './PaginationDots'
 
 import sliderCSS from './slider.css'
 
-const Slider: FC = ({ children }) => {
+const Slider: FC<SliderLayoutSiteEditorProps & { totalItems: number }> = ({
```
Comment (emersonlaurentino) [inline react/components/Slider.tsx]: why do not create a new interface that extends `SliderLayoutSiteEditorProps` and add the `totalItems` as a attribute?

File: react/SliderLayout.tsx [hunk 2]
```diff
@@ -1,14 +1,88 @@
...
+    },
+    showNavigationArrows: {
+      default: 'always',
+      enum: ['mobileOnly', 'desktopOnly', 'always', 'never'],
+      title: messages.sliderShowNavigation,
+      type: 'string',
+    },
+    showPaginationDots: {
+      default: 'always',
+      enum: ['mobileOnly', 'desktopOnly', 'always', 'never'],
```
Comment (fanny) [inline react/SliderLayout.tsx]: A const would be more appropriate, since you need to use in many places

File: react/components/Slider.tsx (hunk 1 above)
Comment (victorhmp) [reply react/components/Slider.tsx]: Nice! Would make it cleaner :)

File: react/SliderLayout.tsx (hunk 2 above)
Comment (victorhmp) [reply react/SliderLayout.tsx]: This array of values is only being used inside this object `SliderLayout.schema`, and just twice. I think it doesn't hurt but actually improves readability if the enum definitions are written like that. If this eventually gets used somewhere else I'll save it in a variable :)

**Pull request merged**